thrift/thrift-gen/main.go

```diff
@@ -149,3 +200,67 @@ func cleanGeneratedCode(generated []byte) []byte {
 func contextType() string {
 	return "thrift.Context"
 }
+
+func getTemplateFromFile(filename string, AST *parser.Thrift) (*template.Template, error) {
+	// export useful functions to the template renderer
+	state := NewState(AST)
+	funcs := map[string]interface{}{
+		"contextType": contextType,
+		"lowercase": func(in string) string {
+			return strings.ToLower(in)
+		},
+		"goPublicName": goPublicName,
+		"goType": func(t *parser.Type) string {
+			return state.goType(t)
+		},
+	}
+
+	// read template file
+	templateBytes, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	templateString := string(templateBytes)
+
+	// parse template in usable form
+	return template.Must(template.
+		New("thrift-gen").
+		Funcs(funcs).
+		Parse(templateString)), nil
+}
+
+func getTemplatesFromFiles(files []string, AST *parser.Thrift) (map[string]*template.Template, error) {
+	var output map[string]*template.Template
+	output = make(map[string]*template.Template)
+
+	for _, filename := range files {
+		parsedTemplate, err := getTemplateFromFile(filename, AST)
+		if err != nil {
+			// PANIC!
+			return nil, err
+		} else {
+			// add parsed template to templates
+			output[filename] = parsedTemplate
+		}
+	}
+	return output, nil
+}
+
+func outputDirectory(infile string, AST *parser.Thrift) string {
```

Comment:
Good catch!

This is probably from one of the earlier prototypes where I got the package name from the AST to get the correct path to write to. After some refactoring that wasn't needed anymore, but apparently I didn't remove it.